src/main/kotlin/racingcar/domain/IOHandler.kt

```diff
@@ -0,0 +1,72 @@
+package racingcar.domain
+
+import camp.nextstep.edu.missionutils.Console
+import racingcar.data.Car
+
+class IOHandler(private val validator: Validator) {
+    fun getCarNames(): List<String> {
+        val input = getInput()
+        val names = splitNameFromInput(input)
+
+        for (name in names) {
+            require(validator.checkCarName(name)) {
+                NAME_LENGTH_IS_NOT_MATCHED
+            }
+        }
+
+        return names
+    }
+
+    fun getMoveCount(): Int {
+        val input = getInput()
+        val count = changeNumStringToInt(input)
+
+        require(validator.checkMoveCount(count)) {
+            NOT_ALLOWED_COUNT
+        }
+
+        return count
+    }
+
+    fun showLocation(cars: List<Car>) {
+        for (car in cars) {
+            show(car)
+        }
+    }
+
+    fun showWinner(winners: List<Car>) {
+        val sentence = winners.joinToString(SEPARATOR_FOR_WINNER) {
+            it.name
+        }
+        show("$SENTENCE_FOR_WINNER$sentence")
+    }
+
+    fun show(sentence: String, lineBreak: Boolean = false) {
+        if (lineBreak) {
+            println(sentence)
+            return
+        }
+        print(sentence)
+    }
+
+    private fun show(car: Car) = println(car)
+
+    private fun getInput(): String = Console.readLine()
+
+    private fun splitNameFromInput(input: String): List<String> = input.split(INPUT_SPLITTER)
+
+    private fun changeNumStringToInt(input: String): Int {
+        return requireNotNull(input.toIntOrNull()) {
+            INPUT_IS_NOT_NUM
+        }
+    }
```

Comment:
[1, 2, 3] 전 아래와 같이 구현을 했었는데 가독성인 면에서 종성님은 어떤 것을 선택하실지 궁금합니다. 전 종성님 코드가 더 좋아보이긴 하네요 ㅎㅎ
```suggestion
    private fun changeNumStringToInt(input: String) =
        input.toIntOrNull() ?: throw IllegalArgumentException(INPUT_IS_NOT_NUM)
```

Reply:
모든 에러 처리를 require로 처리하도록 통일해야겠다고 생각했었는데, 이 원칙을 지키려고 하니 제약사항이 많아지고, IllegalArgumentException이 아닌 경우의 처리가 애매하더라구요. 
 그리고 한 블락의 코드만 실행되므로, 제안 주신 코드처럼 작성하는 게 더 가독성이 좋은 것 같아요. 에러 처리 코드를 작성할 때, 참고하겠습니다!